validator/client/src/test/java/tech/pegasys/teku/validator/client/RuntimeProposerConfigTest.java

```diff
@@ -36,42 +37,69 @@ class RuntimeProposerConfigTest {
     assertThat(proposerConfig.getEth1AddressForPubKey(pubkey)).contains(address);
   }
 
   @Test
-  void shouldUpdateEntries() {
-    proposerConfig.addOrUpdate(pubkey, address);
+  void shouldAddGasLimitEntries() {
+    assertThat(proposerConfig.getGasLimitForPubKey(pubkey)).isEmpty();
+    proposerConfig.addOrUpdateGasLimit(pubkey, gasLimit);
+    assertThat(proposerConfig.getGasLimitForPubKey(pubkey)).contains(gasLimit);
+  }
+
+  @Test
+  void shouldUpdateFeeRecipientEntries() {
+    proposerConfig.addOrUpdateFeeRecipient(pubkey, address);
     assertThat(proposerConfig.getEth1AddressForPubKey(pubkey)).contains(address);
-    proposerConfig.addOrUpdate(pubkey, address2);
+    proposerConfig.addOrUpdateFeeRecipient(pubkey, address2);
     assertThat(proposerConfig.getEth1AddressForPubKey(pubkey)).contains(address2);
   }
 
   @Test
-  void shouldDeleteEntries() {
-    proposerConfig.addOrUpdate(pubkey, address);
+  void shouldUpdateGasLimitEntries() {
+    proposerConfig.addOrUpdateGasLimit(pubkey, gasLimit);
+    assertThat(proposerConfig.getGasLimitForPubKey(pubkey)).contains(gasLimit);
+    proposerConfig.addOrUpdateGasLimit(pubkey, gasLimit2);
+    assertThat(proposerConfig.getGasLimitForPubKey(pubkey)).contains(gasLimit2);
+  }
+
+  @Test
+  void shouldDeleteFeeRecipientEntries() {
+    proposerConfig.addOrUpdateFeeRecipient(pubkey, address);
     assertThat(proposerConfig.getEth1AddressForPubKey(pubkey)).contains(address);
-    proposerConfig.delete(pubkey);
+    proposerConfig.deleteFeeRecipient(pubkey);
     assertThat(proposerConfig.getEth1AddressForPubKey(pubkey)).isEmpty();
   }
 
+  @Test
+  void shouldDeleteGasLimitEntries() {
+    proposerConfig.addOrUpdateGasLimit(pubkey, gasLimit);
+    assertThat(proposerConfig.getGasLimitForPubKey(pubkey)).contains(gasLimit);
+    proposerConfig.deleteGasLimit(pubkey);
+    assertThat(proposerConfig.getGasLimitForPubKey(pubkey)).isEmpty();
+  }
+
   @Test
   void shouldSave(@TempDir final Path tempDir) throws IOException {
     final Path testData = tempDir.resolve("test");
     proposerConfig = new RuntimeProposerConfig(Optional.of(testData));
-    proposerConfig.addOrUpdate(pubkey, address);
-    proposerConfig.addOrUpdate(pubkey2, address2);
+    proposerConfig.addOrUpdateFeeRecipient(pubkey, address);
+    proposerConfig.addOrUpdateGasLimit(pubkey, gasLimit);
+    proposerConfig.addOrUpdateFeeRecipient(pubkey2, address2);
+    proposerConfig.addOrUpdateGasLimit(pubkey2, gasLimit2);
     final String data = Files.readString(testData);
     assertThat(data)
         .isEqualTo(
             String.format(
-                "{\"%s\":{\"fee_recipient\":\"%s\"},\"%s\":{\"fee_recipient\":\"%s\"}}",
-                pubkey, address, pubkey2, address2));
+                "{\"%s\":{\"fee_recipient\":\"%s\",\"gas_limit\":\"%s\"},\"%s\":{\"fee_recipient\":\"%s\",\"gas_limit\":\"%s\"}}",
+                pubkey, address, gasLimit, pubkey2, address2, gasLimit2));
```

Comment:
we could potentially have one test that saves both like here (just to 1 object)
then a second test that stores fee_recipient on one pubkey and gaslimit on a second, and then we have the 3 cases covered...